src/editors/Flow/view/Core/Graph/GraphBase.js

```diff
@@ -571,15 +573,47 @@ export default class GraphBase {
     this.warningsVisibility = isVisible;
   };
 
+  /**
+   * Update node running status
+   *
+   * @param {Object} nodes
+   * @param {*} robotStatus
+   */
   nodeStatusUpdated(nodes) {
+    // Let's wait untill allNodes are loaded to recall this function again
+    // What would happen sometimes is it was called before all nodes were loaded
+    // Causing some nodes to not light up when they should.
+    if (!this.allNodesLoaded)
+      return setTimeout(() => this.nodeStatusUpdated(nodes), 500);
+
     Object.keys(nodes).forEach(nodeName => {
       const status = nodes[nodeName];
-      const node = this.nodes.get(nodeName);
-      if (node)
-        node.obj.status = [1, true, "true"].includes(status) ? true : false;
+      this.updateNodeStatus(nodeName, status);
     });
   }
 
+  /**
+   * @private function
+   * Iterate through the nodes in tree to update its running status
+   *
+   * @param {String} nodeName : Node instance name
+   * @param {Boolean} status : True -> Running / False -> Not Running
+   * @param {TreeContainerNode} parent : Flow to look for the node
+   */
+  updateNodeStatus = (nodeName, status) => {
```

Comment:
why renaming function to remove "_"?

Reply:
Was conventioned a while back (with @rianmartins) that `_` functions should be renamed to the same name but without the `_` and then add the `@private function` to the JSDoc function description.